src/db/notifications.rs

```diff
@@ -139,6 +141,36 @@ pub async fn delete_ping(
                 return Ok(vec![deleted_notification]);
             }
         },
+        Identifier::All => {
+            let rows = db
+                .query(
+                    "DELETE FROM notifications WHERE user_id = $1
+                        RETURNING origin_url, origin_html, time, short_description, metadata",
+                    &[&user_id],
+                )
+                .await
+                .context("delete all notifications query")?;
+            if rows.is_empty() {
+                anyhow::bail!("Did not delete any notifications");
```

Comment:
I don't think this should be an error return -- does our caller not handle an empty vector appropriately?

Reply:
Hmm, I don't know; I just copied the code for `ack`ing a single notification and modified it. I'll look into it.

Reply:
Okay, here's the caller logic:

https://github.com/rust-lang/triagebot/blob/e60ffaddae21a8ad5763e4c8af3750c7151ae422/src/zulip.rs#L490-L511

Basically, if there were no notifications found, it will print:

> Acknowledged:

without any messages, which looks a bit confusing. So I think it might make sense to keep the error. What do you think?

Reply:
Also it seems like maybe it would make sense to set a flag as to whether the notification was acknowledged, rather than deleting it? Although it doesn't matter too much.

Reply:
I think we should change the caller then.

I don't want to keep them around indefinitely.